arquillian-webcam-video-recorder/src/main/java/org/arquillian/extension/recorder/video/webcam/impl/WebcamRecorder.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat Middleware LLC, and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.arquillian.extension.recorder.video.webcam.impl;
+
+import com.github.sarxos.webcam.Webcam;
+import org.arquillian.extension.recorder.video.Video;
+import org.arquillian.extension.recorder.video.VideoConfiguration;
+import org.arquillian.extension.recorder.video.VideoType;
+import org.arquillian.extension.recorder.video.webcam.configuration.WebcamVideoConfiguration;
+import org.jboss.arquillian.core.spi.Validate;
+import org.jcodec.api.awt.SequenceEncoder;
+
+import java.awt.Dimension;
+import java.awt.Toolkit;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ *
+ * @author <a href="mailto:asotobu@gmail.com">Alex Soto</a>
+ */
+class WebcamRecorder {
+
+    private static final Logger logger = Logger.getLogger(WebcamRecorder.class.getName());
+
+    public static final int DEFAULT_FRAMERATE = 20;
+
+    public static final int MIN_FRAMERATE = 0;
+
+    public static final int MAX_FRAMERATE = 100;
+    public static final int ONE_SECOND = 1000;
+
+    private int frameRate = DEFAULT_FRAMERATE;
+
+    private final Dimension screenBounds;
+
+    private volatile boolean running = false;
+
+    private Thread thread;
+
+    private File recordedVideo = null;
+
+    private Timer timer;
+
+    private VideoConfiguration configuration;
+
+    public WebcamRecorder(WebcamVideoConfiguration configuration) {
+        Validate.notNull(configuration, "Video configuration is null!");
+        this.configuration = configuration;
+        this.frameRate = configuration.getFrameRate();
+        screenBounds = Toolkit.getDefaultToolkit().getScreenSize();
```

Comment:
This assumes the webcam res == screen size... Not the case. height:width set in reporter seems to be ignored